pass-fedora-client/src/main/java/org/dataconservancy/pass/client/fedora/FedoraPassCrudClient.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2018 Johns Hopkins University
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.dataconservancy.pass.client.fedora;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.dataconservancy.pass.client.PassJsonAdapter;
+import org.dataconservancy.pass.client.adapter.PassJsonAdapterBasic;
+import org.dataconservancy.pass.model.PassEntity;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.fcrepo.client.DeleteBuilder;
+import org.fcrepo.client.FcrepoClient;
+import org.fcrepo.client.FcrepoOperationFailedException;
+import org.fcrepo.client.FcrepoResponse;
+import org.fcrepo.client.GetBuilder;
+import org.fcrepo.client.PostBuilder;
+import org.fcrepo.client.PutBuilder;
+
+/**
+ * Fedora CRUD client does basic work of creating, retrieving, updating, and deleting
+ * records in Fedora based on model and/or URI provided.
+ * @author Karen Hanson
+ * @version $Id$
+ */
+public class FedoraPassCrudClient {
+
+    private static final Logger LOG = LoggerFactory.getLogger(FedoraPassCrudClient.class);
+
+    private final static String JSONLD_CONTENTTYPE = "application/ld+json; charset=utf-8";
+    private final static String SERVER_MANAGED_OMITTYPE = "http://fedora.info/definitions/v4/repository#ServerManaged";
+    private final static String COMPACTED_ACCEPTTYPE = "application/ld+json; profile=\"http://www.w3.org/ns/json-ld#compacted\"";
+    
+    /** The Fedora client tool */
+    private FcrepoClient client;
+    
+    /** A JSON adapter for PASS */
+    private PassJsonAdapter adapter;
+
+    public FedoraPassCrudClient() {
+        client = FcrepoClient.client().credentials(FedoraConfig.getUserName(), FedoraConfig.getPassword()).build();
+        adapter = new PassJsonAdapterBasic();
+    }
+
+    /** 
+     * Support passing in of Fedora client and adapter
+     * @param client
+     * @param adapter
+     */
+    public FedoraPassCrudClient(FcrepoClient client, PassJsonAdapter adapter) {
+        this.client = client;   
+        this.adapter = adapter;
+        //set defaults if values are null
+        if (this.client == null) {

```

Comment:
If a caller provides `null` arguments, that would argue for an `IllegalArgumentException` to be thrown.  That is, the caller is doing something wrong.  I'd prefer that we not correct bad behavior by setting defaults and throw the IAE instead.